planner/routers/point2point_router.py

```diff
@@ -1,13 +1,48 @@
-from routers.base_router import BaseRouter
+import itertools
+from pprint import pprint
+from typing import *
 
+from routers.base_router import RouteResult
 
-class Point2PointRouter(BaseRouter):
+
+class Point2PointRouter:
     def __init__(self, conn):
         self.conn = conn
 
-    def make_route(self, origin, destination, *args, **kwargs):
+    def make_route(self, origins, dests) -> List[RouteResult]:
+        """
+        :param origins: (lat, lon) pairs.
+        :param dests: (lat, lon) pairs.
+        :return:
+        """
+        output = []
         with self.conn.cursor() as cur:
-            cur.execute('SELECT * FROM pathFromNearestKnownPoints(%s,%s,%s,%s)',
-                        (*origin[::-1], *destination[::-1]))
-            linestring, length = cur.fetchone()
-            return linestring, length
+            for origin, dest in itertools.product(origins, dests):
+                cur.execute(
+                    'SELECT * FROM pathFromNearestKnownPoints(%s,%s,%s,%s)',
+                    (*origin[::-1], *dest[::-1]))
+                linestring, length = cur.fetchone()
+                output.append(RouteResult(
+                    route=linestring,
```

Comment:
This doesn't tell you which origins/destinations are used in the routing?

Reply:
Yeah it doesn't. Though if they were to be specified how would it be done?

Reply:
Maybe start vertex and end vertex, indexed by the origin/destination lists?

For example one-to-one routing will have both start and end vertex id as 0.

Reply:
Technically you can get that by looking at the start and end of the linestring...I kind of want to wait until frontend/web API tells us exactly what they want before implementing it?

Reply:
We also don't return a list of the POIs visited on each route, nor any information about the POIs.

Reply:
Yeah but we have a 'snap to grid' feature in the route planner, so the actual start and end may not be the same points as the origins/destinations provided.